beamer/tests/contracts/test_request_manager.py

```diff
@@ -160,31 +206,61 @@ def test_claim_period_extension(
     claim = request_manager.claimRequest(request_id, 0, {"from": claimer, "value": claim_stake})
     claim_id = claim.return_value
 
-    assert claim.timestamp + claim_period == request_manager.claims(claim_id)[6]
+    assert (
+        claim.timestamp + claim_period == request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION]
+    )
 
     challenge = request_manager.challengeClaim(
         claim_id, {"from": challenger, "value": claim_stake + 1}
     )
     challenge_period = finalization_time + challenge_period_extension
 
-    assert challenge.timestamp + challenge_period == request_manager.claims(claim_id)[6]
+    assert (
+        challenge.timestamp + challenge_period
+        == request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION]
+    )
 
     # Another challenge with big margin to the end of the termination
     # shouldn't increase the termination
     request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
-    assert challenge.timestamp + challenge_period == request_manager.claims(claim_id)[6]
+    assert (
+        challenge.timestamp + challenge_period
+        == request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION]
+    )
+
+    # Another challenge by challenger also shouldn't increase the end of termination
+    request_manager.challengeClaim(claim_id, {"from": challenger, "value": claim_stake + 1})
+    assert (
+        challenge.timestamp + challenge_period
+        == request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION]
+    )
 
     # Timetravel close to end of challenge period
-    chain.mine(timedelta=challenge_period * 8 / 10)
+    chain.mine(timestamp=request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION] - 10)
+
+    # Claimer challenges close to the end of challenge
+    # Should increase the challenge termination
+    challenge = request_manager.challengeClaim(
+        claim_id, {"from": claimer, "value": claim_stake + 1}
+    )
+    assert (
+        challenge.timestamp + challenge_period_extension
+        == request_manager.claims(claim_id)[RM_C_FIELD_TERMINATION]
+    )
```

Comment:
Can we also check here that the new termination is strictly greater than the old one?